prompting/tasks/task_registry.py

```diff
@@ -0,0 +1,66 @@
+from prompting.tasks.base_task import BaseTask
+from prompting.rewards.reward import BaseRewardConfig
+from prompting.tasks.date_qa import DateQuestionAnsweringTask, DateQARewardConfig
+from prompting.tasks.qa import QuestionAnsweringTask, QARewardConfig
+from prompting.tasks.summarization import SummarizationTask, SummarizationRewardConfig
+
+from prompting.datasets.wiki import WikiDataset, WikiDateDataset
+from prompting.datasets.base import BaseDataset
+from pydantic import BaseModel, ConfigDict
+import random
+from typing import ClassVar
+from loguru import logger
+
+
+class TaskConfig(BaseModel):
+    task: BaseTask.__class__
+    probability: float
+    datasets: list[BaseDataset.__class__]
+    reward_model: BaseRewardConfig.__class__
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+
+class TaskRegistry(BaseModel):
+    task_configs: ClassVar[list[TaskConfig]] = [
+        TaskConfig(task=QuestionAnsweringTask, probability=0.6, datasets=[WikiDataset], reward_model=QARewardConfig),
+        TaskConfig(
+            task=SummarizationTask, probability=0.2, datasets=[WikiDataset], reward_model=SummarizationRewardConfig
+        ),
+        TaskConfig(
+            task=DateQuestionAnsweringTask, probability=0.2, datasets=[WikiDateDataset], reward_model=DateQARewardConfig
+        ),
+    ]
```

Comment:
We should move task probabilities to config file

Reply:
Is there a good reason why? They should never be changed on the validator side, right

Reply:
Resolved based on conversation - let's review in future how best to do config

Reply:
just to make it more future proof to changes, if the project will grow to multiple files like: `task_registry, abc.py, efg.py` each having it’s own values, then it’s easier to modify config params in constants.py or config